datasets/epa_historical_air_quality/_images/run_csv_transform_kub/csv_transform.py

```diff
@@ -0,0 +1,292 @@
+        encoding="utf-8",
+        quotechar='"',  # string separator, typically double-quotes
+        chunksize=chunksize,  # size of batch data, in no. of records
+        sep=",",  # data column separator, typically ","
+        header=None,  # use when the data file does not contain a header
+        names=names,
+        dtype=dtypes,
+        keep_default_na=True,
+        na_values=[" "],
+    ) as reader:
+        for chunk_number, chunk in enumerate(reader):
+            target_file_batch = str(target_file).replace(
+                ".csv", "-" + str(chunk_number) + ".csv"
+            )
+            df = pd.DataFrame()
+            df = pd.concat([df, chunk])
+            process_chunk(df, target_file_batch, target_file, (not chunk_number == 0))
+
+
+def process_chunk(
+    df: pd.DataFrame,
+    target_file_batch: str,
+    target_file: str,
+    skip_header: bool,
+) -> None:
+    df = resolve_date_format(df, "%Y-%m-%d %H:%M")
+    save_to_new_file(df, file_path=str(target_file_batch), sep=",")
+    append_batch_file(target_file_batch, target_file, skip_header, not (skip_header))
+
+
+def resolve_date_format(df: pd.DataFrame, from_format: str) -> pd.DataFrame:
+    logging.info("Resolving Date Format")
+    for col in df.columns:
+        if df[col].dtype == "datetime64[ns]":
+            logging.info(f"Resolving datetime on {col}")
+            df[col] = df[col].apply(lambda x: convert_dt_format(str(x), from_format))
+
+    return df
+
+
+def convert_dt_format(dt_str: str, from_format: str) -> str:
+    # rtnval = ""
+    if not dt_str or str(dt_str).lower() == "nan" or str(dt_str).lower() == "nat":
+        rtnval = ""
+    elif len(dt_str.strip()) == 10:
+        # if there is no time format
+        rtnval = dt_str + " 00:00:00"
+    elif len(dt_str.strip().split(" ")[1]) == 8:
+        # if format of time portion is 00:00:00 then use 00:00 format
+        dt_str = dt_str[:-3]
+        rtnval = datetime.datetime.strptime(dt_str, from_format).strftime(
+            "%Y-%m-%d %H:%M:%S"
+        )
+    elif (len(dt_str.strip().split("-")[0]) == 4) and (
+        len(from_format.strip().split("/")[0]) == 2
+    ):
+        # if the format of the date portion of the data is in YYYY-MM-DD format
+        # and from_format is in MM-DD-YYYY then resolve this by modifying the from_format
+        # to use the YYYY-MM-DD.  This resolves mixed date formats in files
+        from_format = "%Y-%m-%d " + from_format.strip().split(" ")[1]
+    else:
+        dt_str = ""
+
+    # return datetime.datetime.strptime(dt_str, from_format).strftime("%Y-%m-%d %H:%M:%S")
+    return rtnval
+
+
+def save_to_new_file(df, file_path, sep="|") -> None:
+    logging.info(f"Saving to file {file_path} separator='{sep}'")
+    df.to_csv(file_path, sep=sep, index=False)
+
+
+def append_batch_file(
+    batch_file_path: str, target_file_path: str, skip_header: bool, truncate_file: bool
+) -> None:
+    data_file = open(batch_file_path, "r")
```

Comment:
Another way to do this is using context blocks:
```
with open(batch_file_path, "r") as data_file:
   # code here
```
This prevents from having to manually close the file via `data_file.close()`.

Reply:
Done.